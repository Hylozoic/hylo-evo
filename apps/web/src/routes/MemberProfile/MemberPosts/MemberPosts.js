import React from 'react'
import Loading from 'components/Loading'
import PostCard from 'components/PostCard'
import classes from './MemberPosts.module.scss'

export default class MemberPosts extends React.Component {
  static defaultProps = {
    routeParams: {}
  }

  componentDidMount () {
    this.props.fetchMemberPosts()
  }

  itemSelected = selectedItemId => selectedItemId === this.props.routeParams.postId

  render () {
    if (this.props.loading) return <Loading />

    const { posts, routeParams } = this.props

    return <div>
      {posts && posts.map(post =>
        <div className={classes.activityItem} key={post.id}>
          <PostCard
            routeParams={routeParams}
            post={post}
            expanded={this.itemSelected(post.id)} />
        </div>
      )}
    </div>
  }
}
