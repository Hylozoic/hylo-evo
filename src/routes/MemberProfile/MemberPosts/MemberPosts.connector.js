import { connect } from 'react-redux'
import { push } from 'react-router-redux'

import { memberPostsSelector, fetchMemberPosts } from './MemberPosts.store'

export function mapStateToProps (state, props) {
  return {
    posts: memberPostsSelector(state, props)
  }
}

export const mapDispatchToProps = {
  fetchMemberPosts,
  showDetails: (id, slug) => push(`/c/${slug}/p/${id}`)
}

export default connect(mapStateToProps, mapDispatchToProps)
