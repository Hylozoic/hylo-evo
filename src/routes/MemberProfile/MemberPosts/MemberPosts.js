import PropTypes from 'prop-types'
import React from 'react'
import { get } from 'lodash/fp'
import PostCard from 'components/PostCard'
import './MemberPosts.scss'

const { any, arrayOf, func, number, shape, string } = PropTypes

const personShape = shape({
  id: any,
  name: string,
  avatarUrl: string
})
const communityShape = shape({
  id: any,
  name: string,
  slug: string
})

export default class MemberPosts extends React.Component {
  static propTypes = {
    personId: string,
    posts: arrayOf(shape({
      id: any,
      commenters: arrayOf(personShape),
      communities: arrayOf(communityShape),
      commentersTotal: number,
      creator: personShape,
      createdAt: string,
      details: string,
      followers: arrayOf(personShape),
      title: string,
      type: string
    })),
    showDetails: func,
    editPost: func
  }

  componentDidMount () {
    this.props.fetchMemberPosts(this.props.personId)
  }

  render () {
    const { personId, posts, showDetails, editPost } = this.props
    return <div>
      {posts && posts.map(post => {
        const slug = get('communities.0.slug', post)
        return <div styleName='activity-item' key={post.id}>
          <PostCard
            post={post}
            showDetails={() => showDetails(post.id, slug, personId)}
            editPost={() => editPost(post.id, slug, personId)} />
        </div>
      }

      )}
    </div>
  }
}
