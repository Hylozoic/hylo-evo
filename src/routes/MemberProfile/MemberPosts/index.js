import MemberPosts from './MemberPosts'
import connector from './MemberPosts.connector'
export default connector(MemberPosts)
