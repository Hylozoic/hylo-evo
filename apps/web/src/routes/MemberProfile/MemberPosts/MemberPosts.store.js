import { createSelector as ormCreateSelector } from 'redux-orm'
import orm from 'store/models'
import presentPost from 'store/presenters/presentPost'
import postsQueryFragment from '@graphql/fragments/postsQueryFragment'
import { FETCH_MEMBER_POSTS } from '../MemberProfile.store'

const memberPostsQuery =
`query MemberPosts (
  $activePostsOnly: Boolean,
  $afterTime: Date,
  $announcementsOnly: Boolean,
  $beforeTime: Date,
  $boundingBox: [PointInput],
  $collectionToFilterOut: ID,
  $context: String,
  $createdBy: [ID],
  $cursor: ID,
  $filter: String,
  $first: Int,
  $forCollection: ID,
  $groupSlugs: [String],
  $id: ID,
  $interactedWithBy: [ID],
  $isFulfilled: Boolean,
  $mentionsOf: [ID],
  $offset: Int,
  $order: String,
  $search: String,
  $sortBy: String,
  $topic: ID,
  $topics: [ID],
  $types: [String]
) {
  person (id: $id) {
    id
    ${postsQueryFragment}
  }
}`

export function fetchMemberPosts (id, first = 20, query = memberPostsQuery) {
  return {
    type: FETCH_MEMBER_POSTS,
    graphql: {
      query,
      variables: { id, first }
    },
    meta: { extractModel: 'Person' }
  }
}

export const getMemberPosts = ormCreateSelector(
  orm,
  (_, { routeParams }) => routeParams,
  ({ Person }, { personId }) => {
    if (!Person.idExists(personId)) return
    return Person.withId(personId).posts.toModelArray().map(post =>
      presentPost(post))
  }
)
