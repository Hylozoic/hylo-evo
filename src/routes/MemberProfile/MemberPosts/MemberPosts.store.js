import { createSelector } from 'redux-orm'
import orm from 'store/models'
import { FETCH_MEMBER_POSTS } from '../MemberProfile.store'
import { postsQueryFragment } from 'components/FeedList/FeedList.store'

const memberPostsQuery =
`query MemberPosts (
  $id: ID,
  $sortBy: String,
  $offset: Int,
  $search: String,
  $filter: String,
  $first: Int,
  $topic: Int
) {
  person (id: $id) {
    id
    ${postsQueryFragment}
  }
}`

export function fetchMemberPosts (id, first = 20, query = memberPostsQuery) {
  return {
    type: FETCH_MEMBER_POSTS,
    graphql: {
      query,
      variables: {id, first}
    },
    meta: {extractModel: 'Person'}
  }
}

export const memberPostsSelector = createSelector(
  orm,
  state => state.orm,
  (_, { personId }) => personId,
  (_, { slug }) => slug,
  (session, personId, slug) => {
    if (session.Person.hasId(personId)) {
      const person = session.Person.withId(personId)
      return person.posts.toModelArray().map(post => ({
        ...post.ref,
        creator: post.creator.ref,
        linkPreview: post.linkPreview,
        commenters: post.commenters.toRefArray(),
        communities: post.communities.toRefArray(),
        fileAttachments: post.attachments.filter(a => a.type === 'file').toModelArray()
      }))
    }
    return null
  }
)
