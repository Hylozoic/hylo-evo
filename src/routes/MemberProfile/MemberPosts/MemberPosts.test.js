import { shallow } from 'enzyme'
import React from 'react'

import MemberPosts from './MemberPosts'
import denormalized from '../MemberProfile.test.json'

describe.only('MemberPosts', () => {
  const { person } = denormalized.data

  it('renders the same as the last snapshot', () => {
    const wrapper = shallow(
      <MemberPosts fetchMemberPosts={jest.fn()} posts={person.posts} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
