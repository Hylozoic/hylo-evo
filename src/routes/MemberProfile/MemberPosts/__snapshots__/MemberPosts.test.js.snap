// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberPosts renders the same as the last snapshot 1`] = `
<div>
  <div
    data-stylename="activity-item"
    key="30003"
  >
    <PostCard
      editPost={[Function]}
      post={
        Object {
          "commenters": Array [],
          "commentersTotal": 0,
          "communities": Array [
            Object {
              "id": "1836",
              "name": "Wombats",
              "slug": "wombats",
            },
          ],
          "createdAt": "Sat Apr 15 2017 16:27:59 GMT+1200 (NZST)",
          "creator": Object {
            "id": "46816",
          },
          "details": "<p>With nothing of importance in it.</p>",
          "id": "30003",
          "title": "Some other post",
          "type": "discussion",
        }
      }
      showDetails={[Function]}
    />
  </div>
  <div
    data-stylename="activity-item"
    key="30002"
  >
    <PostCard
      editPost={[Function]}
      post={
        Object {
          "commenters": Array [
            Object {
              "avatarUrl": "https://www.gravatar.com/avatar/32f7ff34c3b5a8600b79be9a85a5c92f?d=mm&s=140",
              "id": "46816",
              "name": "Rich Churcher",
            },
          ],
          "commentersTotal": 1,
          "communities": Array [
            Object {
              "id": "1836",
              "name": "Wombats",
              "slug": "slug",
            },
          ],
          "createdAt": "Sat Mar 18 2017 10:48:43 GMT+1300 (NZDT)",
          "creator": Object {
            "id": "46816",
          },
          "details": "<p><a href=\\"https://wombat.life\\">https://wombat.life</a></p>
<p></p>
<p></p>",
          "id": "30002",
          "title": "0123456789 0123456789 0123456789",
          "type": "discussion",
        }
      }
      showDetails={[Function]}
    />
  </div>
</div>
`;
